python/cudf/cudf/core/single_column_frame.py

```diff
@@ -216,9 +217,9 @@ def is_monotonic_increasing(self):
         """
         return self._column.is_monotonic_increasing
 
-    @property  # type: ignore
+    @cached_property
     @_cudf_nvtx_annotate
-    def is_monotonic_decreasing(self):
+    def is_monotonic_decreasing(self) -> bool:
```

Comment:
Similarly here, we're allowed to "swap out" the column and leave the same `Frame` object around (see `Frame._mimic_inplace`), so without invalidation, it's unsafe to make these objects cached properties.